src/rf/js/src/home/components/modals.js

```diff
@@ -107,9 +188,224 @@ var UploadModal = React.createBackboneClass({
+                capture_end: getTextValue('capture_end'),
+                area: getTextValue('area'),
+                area_units: getTextValue('area_units'),
+                projection: '4325', //getSelectValue('projection'),
+                srid: '4325', //getTextValue('srid'),
+                tile_srid: getSelectValue('tile_srid'),
+                tile_format: getSelectValue('tile_format'),
+                resampling: getSelectValue('resampling'),
+                transparency: getChecked('transparency') ?
+                    defaultColor : getTextValue('transparency_color'),
+                tile_origin: getChecked('tile_origin') ?
+                    'topleft' : 'bottomleft'
+            };
+
+        return layerFields;
+    },
+
+    getClientErrors: function() {
+        var errors = this.getInitialClientErrors();
+
+        if (this.state.fileDescriptions.length === 0) {
+            errors.files = 'Please select at least one file.';
+        }
+
+        if (!this.state.name) {
+            errors.name = 'Please enter a name.';
+        } else if (this.state.name.length > 255) {
+            errors.name = 'Please enter a name < 255 characters.';
+        }
+
+        if (!this.state.organization) {
+            errors.organization = 'Please enter an organization.';
+        } else if (this.state.organization.length > 255) {
+            errors.organization = 'Please enter an organization < 255 characters.';
+        }
+
+        if (!this.state.description) {
+            errors.description = 'Please enter a description.';
+        }
+
+        function setDateErrors(state, errors, fieldName) {
+            if (!state[fieldName]) {
+                errors[fieldName] = 'Please enter a valid date.';
+            } else {
+                var date = moment(state[fieldName]);
+                if (date.year() < 1800) {
+                    errors[fieldName] = 'Please enter a year after 1500';
+                } else if (date.year() > 2500) {
+                    errors[fieldName] = 'Please enter a year before 2500';
+                }
+            }
+        }
+
+        setDateErrors(this.state, errors, 'capture_start');
+        setDateErrors(this.state, errors, 'capture_end');
+
+        if(this.state.capture_start && this.state.capture_end) {
+            var start = moment(this.state.capture_start),
+                end = moment(this.state.capture_end);
+            if (end.isBefore(start)) {
+                errors.capture_dates = 'Please enter a capture end that is not before the start.';
+            }
+        }
+
+        if (!this.state.area) {
+            errors.area = 'Please enter an area.';
+        }
+
+        return errors;
+    },
+
+    getServerErrors: function(responseJSON) {
+        var errors = this.getInitialServerErrors();
+
+        if (responseJSON.errors.duplicate_name) {
+            errors.duplicate_name = 'Please choose a name not already in use.';
```

Comment:
Minor, but the server-side validation returns an error message that gets discarded here.